src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+use std::{
+    cmp::Ordering,
+    io::{self, BufRead},
+};
+
+use log::warn;
+
+use crate::collapse::{common::Occurrences, Collapse};
+
+static START_LINE: &str = "Level,Function Name,Number of Calls,Elapsed Inclusive Time %,Elapsed Exclusive Time %,Avg Elapsed Inclusive Time,Avg Elapsed Exclusive Time,Module Name,";
+
+/// A stack collapser for the output of the built in Visual Studio profiler.
+#[derive(Default)]
+pub struct Folder {
+    /// Function entries on the stack in this entry thus far.
+    stack: Vec<(String, usize)>,
+}
+
+impl Collapse for Folder {
```

Comment:
If you want a slight challenge (definitely not required for the PR to land!), try implementing the `CollapsePrivate` trait instead:
https://github.com/jonhoo/inferno/blob/410018b46983b1ee732f8e6f7f94f843b9de09d5/src/collapse/common.rs#L53-L59

That will give you multi-core file processing, which can speed up collapsing quite a lot.